vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -32,44 +75,55 @@
             datepicker.$connector = {};
 
             /* init helper parts for reverse-engineering date-regex */
-            datepicker.$connector.dayPart = new FlowDatePickerPart("22");
-            datepicker.$connector.monthPart = new FlowDatePickerPart("11");
-            datepicker.$connector.yearPart = new FlowDatePickerPart("1987");
-            datepicker.$connector.parts = [datepicker.$connector.dayPart, datepicker.$connector.monthPart, datepicker.$connector.yearPart];
+            const dayPart = datepicker.$connector.dayPart = new FlowDatePickerPart("22");
+            const monthPart = datepicker.$connector.monthPart = new FlowDatePickerPart("11");
+            const yearPart = datepicker.$connector.yearPart = new FlowDatePickerPart("1987");
+            const parts = datepicker.$connector.parts = [dayPart, monthPart, yearPart];
 
             // Old locale should always be the default vaadin-date-picker component
             // locale {English/US} as we init lazily and the date-picker formats
             // the date using the default i18n settings and we need to use the input
             // value as we may need to parse user input so we can't use the _selectedDate value.
             let oldLocale = "en-us";
 
-            datepicker.addEventListener('blur', tryCatchWrapper(e => {
+            datepicker.addEventListener('blur', tryCatchWrapper(/** @param {EventWithDatePickerTarget} e */ e => {
                 if (!e.target.value && e.target.invalid) {
                     console.warn("Invalid value in the DatePicker.");
                 }
             }));
 
-            const cleanString = tryCatchWrapper(function (string) {
-                // Clear any non ascii characters from the date string,
-                // mainly the LEFT-TO-RIGHT MARK.
-                // This is a problem for many Microsoft browsers where `toLocaleDateString`
-                // adds the LEFT-TO-RIGHT MARK see https://en.wikipedia.org/wiki/Left-to-right_mark
+            /**
+             * Clear any non ascii characters from the date string,
+             * mainly the LEFT-TO-RIGHT MARK.
+             * This is a problem for many Microsoft browsers where `toLocaleDateString`
+             * adds the LEFT-TO-RIGHT MARK see https://en.wikipedia.org/wiki/Left-to-right_mark
+             * @param {string} string
+             * @returns {string}
+             */
+            let cleanString = function (string) {
                 return string.replace(/[^\x00-\x7F]/g, "");
-            });
+            };
+            cleanString = tryCatchWrapper(cleanString);
```

Comment:
Most `tryCatchWrapper` calls were changed to this style so that the clean function definition is first, and then it is overridden with a wrapped version. This was the only way I could find to make IntelliJ IDEA and VS Code to correctly use the JSDoc for the wrapped function so that the parameter types and return types are correctly available both inside the function and when it's being called somewhere else.